docs/_spec/08-pattern-matching.md

```diff
@@ -307,6 +309,46 @@ All alternative patterns are type checked with the expected type of the pattern.
 They may not bind variables other than wildcards.
 The alternative pattern matches a value ´v´ if at least one its alternatives matches ´v´.
 
+### Given patterns
+
+```ebnf
+  GivenPattern   ::= TypePat
+```
+
+A _given pattern_ introduces an _anonymous given instance_ of the _type_ matched by `TypePat` to the scope of the clause that a given pattern occurs _within_.
```

Comment:
I don't understand what the last part of the sentence means, starting from "to the scope". It does not appear to be grammatically correct, or it's grammatically ambiguous at best. Rephrase?

Reply:
What if the corresponding scrutinee is not statically known to have the type `TypePat`? Is that a compile error or a no-match at run-time?

Reply:
@sjrd I was trying to refer to the scope of the case clause - is the term local scope fine?

I think that in the case that the scrutinee is known statically or not statically should be the same as for a `TypePat` without a `given` modifier.

Currently,

```scala
val x = 1
val y = x match {
  case y: String => ???
}

```

gives a compile error and 

```scala
val x = "asd"

def foo[A](a: A): Unit = {
  a match {
    case s: Int=> 1
  }
}

foo(x)
```

Is a runtime error.